src/docks/encodedock.cpp

```diff
@@ -1389,6 +1389,11 @@ void EncodeDock::on_encodeButton_clicked()
         return;
     }
     bool seekable = MLT.isSeekable(fromProducer());
+
+    if (seekable && checkForMissingFiles()) {
```

Comment:
Possibly instead of creating a new temporary XML file early here you can integrate with the XML file generated in `createMeltJob()`. I think a drawback is repetition when using From = Each Playlist Item. You can change that logic to break out of loop if `createMeltJob` returns null.

Reply:
Yes, I considered that. I did not do that because:
1) I do not even want to prompt the user for a filename if there are missing files
I am open minded about this, but it seems more convenient to me
2) "From=each playlist item" as you mentioned. I could detect the missing file in the loop and bail out. But my message to the user is to close and reopen. They can't do that if 20 encode jobs have just been queued up but the 21st had an error. But maybe that is still a better solution.

I will defer to your preference on this.

Reply:
We will keep it as you have it.